src/intervals/interval_operations/pointwise_boolean.jl

```diff
@@ -0,0 +1,204 @@
+# This file is part of the IntervalArithmetic.jl package; MIT licensed
+
+#=  This file contains functions not specified in the IEEE Std 1788-2015,
+    extending boolean operations defined for reals to intervals.
+
+    This is different from what is described in sections 9.5 of the
+    IEEE Std 1788-2015 (Boolean functions of intervals), since we deal here
+    with what is needed for drop-in replacement of floats by intervals,
+    and not what is reasonnable to do with intervals in a separate execution
+    environment.
+
+    Essentially, julia is better at composability and we are more ambitious
+    that what the standard ever expected, so we have some extra problems
+    to solve.
+
+    We use a trait system, defining the operation with an extra
+    `PointwisePolitic` argument defining how it should be handled.
+
+    The default operations use `IntervalArithmetic.pointwise_politic()`,
+    so in order to change the default behavior this function has to be
+    redefined.
+
+    We define every operations using the default `PointwisePolitic{:ternary}()`
+    and then use it to define all the others.
+=#
+"""
+    PointwisePolitic{P}
+
+Define which politic we use to extend pointwise comparison of 
+
+Valid value for the politic identifier `P` are
+    - `:is_all` : A boolean operation is extended by asking "is it true for
+        all elements of the interval(s) involved".
+        This is self-consistent, but breaks the usual rules for negation.
+        For example with this politic, `((-1..3) == 0) == false` because it
+        answers the question "are all elements in (-1..3) equal to zero".
+        However its negation is `false` too: `((-1..3) != 0) == false`.
+        This *silently* breaks any code relying on such operation in conditional
+        statements like `if x == 0 ... else ... end`.
```

Comment:
I think I the point you want to make is well taken, but I think the example or the wording, are not quite correct. Negating `((-1..3) == 0)` corresponds to `!((-1..3) == 0)` which in this case is not `((-1..3) != 0)`.

Reply:
I tried to make it clearer.